migtests/scripts/resumption.py

```diff
@@ -0,0 +1,314 @@
+    target_db_schema = os.getenv('TARGET_DB_SCHEMA', '')
+    target_db_name = os.getenv('TARGET_DB_NAME', '')
+    data_dir = os.getenv('DATA_DIR', '')
+
+    # Adaptive parallelism
+    run_without_adaptive_parallelism = os.getenv('RUN_WITHOUT_ADAPTIVE_PARALLELISM') == 'true'
+    source_db_type = os.getenv('SOURCE_DB_TYPE', '')
+
+
+def prepare_import_data_file_command():
+    """Prepares the yb-voyager import data file command."""
+    args = [
+        'yb-voyager', 'import', 'data', 'file',
+        '--export-dir', export_dir,
+        '--target-db-host', target_db_host,
+        '--target-db-port', target_db_port,
+        '--target-db-user', target_db_user,
+        '--target-db-password', target_db_password,
+        '--target-db-schema', target_db_schema,
+        '--target-db-name', target_db_name,
+        '--disable-pb', 'true',
+        '--send-diagnostics', 'false',
+        '--data-dir', data_dir,
+        '--file-table-map', file_table_map
+    ]
+
+    if run_without_adaptive_parallelism:
+        args.extend(['--enable-adaptive-parallelism', 'false'])
+
+    for flag, value in additional_flags.items():
+        args.append(flag)
+        args.append(value)
+
+    return args
+
+
+def prepare_import_data_command(config):
+    """
+    Prepares the yb-voyager import data command based on the given configuration.
+    """
+
+    args = [
+        'yb-voyager', 'import', 'data',
+        '--export-dir', export_dir,
+        '--target-db-host', target_db_host,
+        '--target-db-port', target_db_port,
+        '--target-db-user', target_db_user,
+        '--target-db-password', target_db_password,
+        '--target-db-name', target_db_name,
+        '--disable-pb', 'true',
+        '--send-diagnostics', 'false',
+    ]
+    
+    if source_db_type != 'postgresql':
+        args.extend(['--target-db-schema', target_db_schema])
+
+    if run_without_adaptive_parallelism:
+        args.extend(['--enable-adaptive-parallelism', 'false'])
+
+    for flag, value in additional_flags.items():
+        args.append(flag)
+        args.append(value)
+
+    return args
+
+def run_command(command, allow_interruption=False, interrupt_after=None):
+    with tempfile.TemporaryFile() as stdout_file, tempfile.TemporaryFile() as stderr_file:
+        process = subprocess.Popen(
+            command, stdout=stdout_file, stderr=stderr_file, text=True
+        )
+        start_time = time.time()
+        interrupted = False
+
+        while process.poll() is None:
+            if allow_interruption and interrupt_after is not None:
+                elapsed_time = time.time() - start_time
+                if elapsed_time > interrupt_after:
+                    print("Interrupting the process (PID: {})...".format(process.pid), flush=True)
+                    try:
+                        process.terminate()
+                        process.wait(timeout=10)
+                    except subprocess.TimeoutExpired:
```

Comment:
let's put some logs/prints to determine if we terminated the process or killed it